packages/smart-contracts/src/contracts/ChainlinkConversionPath.sol

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.4.25 <0.7.0;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/access/roles/WhitelistAdminRole.sol";
+
+interface ERC20fraction {
+  function decimals() external view returns (uint8);
+}
+
+interface AggregatorFraction {
+  function decimals() external view returns (uint8);
+  function latestAnswer() external view returns (int256);
+  function latestTimestamp() external view returns (uint256);
+}
+
+
+/**
+ * @title ChainlinkConversionPath
+ *
+ * @notice ChainlinkConversionPath is a contract allowing to compute conversion from a path of Chainlink
+ */
+contract ChainlinkConversionPath is WhitelistAdminRole {
+  using SafeMath for uint256;
+
+  uint constant DECIMALS = 1e18;
+
+  // Mapping of Chainlink aggrefators (input currency => output currency => contract address)
+  // input & output currencies are the addresses of the ERC20 contracts OR the sha3("currency code")
+  mapping(address => mapping(address => address)) public allAggregators;
+
+  // rate must have been updated before the last... 10min
+  uint256 public maxTimestampDeltaAcceptable = 600;
```

Comment:
can this become an issue? when ethereum is cluttered for instance

Reply:
There is a choice here: when the network is cluttered, should we reduce the risk of abuse (and UX) by forcing the payer to wait or the opposite? 
I think that at the contract level we should keep a trust-base logic, and maybe this threshold should be set in upper layers.

Reply:
This constant is actually not used. I forgot to remove it.

This mechanism will be set up on the next contract. 

I will open a discussion about it.